### PR TITLE
Processing ActionResult<T> properly. Fixes #4369

### LINKED ISSUE #4369: Issue introduced with version 2.0.12408.0 of the runtime which was not present with 2.0.12382.0
I have detailed the issue here...
https://stackoverflow.com/questions/55839514/intermittent-dependency-injection-failure/55847358#55847358

### PR DESCRIPTION
**Pull request by alrod**

### CODE REVIEW

**Comment by alrod** [comment]
https://github.com/Azure/azure-functions-host/issues/4396

File: src/WebJobs.Script/Binding/Http/HttpBinding.cs [hunk 1]
```diff
@@ -197,6 +198,11 @@ internal static void SetResponse(HttpRequest request, object result)
                 {
                     actionResult = CreateResult(request, result);
                 }
+                else if (result != null && result.GetType().GetInterface("IConvertToActionResult") != null)
```
Comment (fabiocav) [inline src/WebJobs.Script/Binding/Http/HttpBinding.cs]:
```suggestion
  else if (result is IConvertToActionResult convertible)
                {
                    // Convert ActionResult<T> to ActionResult
                    actionResult = convertible.Convert();
                }
```

File: src/WebJobs.Script/Binding/Http/HttpBinding.cs [hunk 2]
```diff
@@ -209,6 +211,11 @@ internal static void SetResponse(HttpRequest request, object result)
                 {
                     actionResult = CreateResult(request, result);
                 }
+                else if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableActionResult) && result is IConvertToActionResult convertable)
```
Comment (fabiocav) [inline src/WebJobs.Script/Binding/Http/HttpBinding.cs]: Can you please move this check to the constructor and cache the result?

File: src/WebJobs.Script/ScriptConstants.cs [hunk 3]
```diff
@@ -90,6 +90,7 @@ public static class ScriptConstants
 
         public const string FeatureFlagDisableShadowCopy = "DisableShadowCopy";
         public const string FeatureFlagsEnableDynamicExtensionLoading = "EnableDynamicExtensionLoading";
+        public const string FeatureFlagEnableActionResult = "FeatureFlagEnableActionResult";
```
Comment (fabiocav) [inline src/WebJobs.Script/ScriptConstants.cs]: The string value shouldn't have the `FeatureFlag` prefix. I suggest something like `EnableActionResultHandling`

**Pull request merged**